workers/workers_test.go

```diff
@@ -131,3 +135,180 @@ func BenchmarkSerial(b *testing.B) {
 		})
 	}
 }
+
+func TestJobWait(t *testing.T) {
```

Comment:
This test is very tuned to the internals of the implementation, usually you would not do this to test a package (makes refactor tough to do).

Reply:
Your "gut check" for this sort of thing should be: "am I using the provided initializer or am I creating the struct from scratch to get a test to run"

Reply:
so Job is an internal struct used by Workers. Basically you're saying we never use Job by itself so theres no point writing tests for it?

Reply:
Not that there isn't a "point per se" but generally people avoid writing tests for internal structs because it makes refactors much more difficult (as you usually try to maintain the same functional interface with changes behind the scenes -> AKA invariants/expectations typically only apply to the public interface not anything internal).

In this case, we can keep it because it is so simple but just a heads up.